app/models/airline.rb

```diff
@@ -1,3 +1,7 @@
 class Airline < ApplicationRecord
+  belongs_to :user, foreign_key: 'user_id'
   has_many :reservations, dependent: :destroy
```

Comment:
Nice!

Although it's not part of the requirements you were able to find a way to delete all reservations related to a deleted airline.